v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+   "source": [
+    "backtest_automl = pd.read_csv(\n",
+    "    os.path.join(OUTPUT_DIR, \"test-set-predictions-many-models.csv\"),\n",
+    "    parse_dates=[time_column_name],\n",
+    ")\n",
+    "backtest_base = pd.read_csv(\n",
+    "    os.path.join(OUTPUT_DIR, \"test-set-predictions-base.csv\"),\n",
+    "    parse_dates=[time_column_name],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Nex, we combine outputs into a single dataframe which will be used for plotting and scoring."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "backtest = backtest_automl.merge(\n",
+    "    backtest_base.drop(target_column_name, axis=1),\n",
+    "    on=[\"customer_id\", \"datetime\"],\n",
+    "    how=\"inner\",\n",
+    "    suffixes=[\"\", \"_base\"],\n",
+    ")\n",
+    "backtest.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\n",
+    "    f\"N model: {backtest_automl.shape[0]}. N baseline: {backtest_base.shape[0]}. N merged: {backtest.shape[0]}\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The data we are working with has an hourly frequency and we plan to generate the forecasts every 24 hours. If the model were to be put in production such that the forecasts are generated and model's performance is monitored every 24 hours, we will mimic the scoring process on the test set by generating daily accuracy metrics. To do this, we create a date column (\"ymd\"). If you want to score the output at any other frequency, say, weekly, just change the frequency parameter to the desired frequency."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PERIOD_COLUMN = \"ymd\"\n",
+    "\n",
+    "backtest[PERIOD_COLUMN] = backtest[time_column_name].dt.to_period(\n",
+    "    \"D\"\n",
+    ")  # year-month-day to be used for daily metrics computation\n",
+    "backtest.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 7.2 Generate time series plots\n",
+    "\n",
+    "Here, we generate forecast versus actuals plot for the test set for both the best many models and the baseline. Since we use rolling evaluation with the step size of 24 hours, this mimics the behavior of putting both models in production and monitoring their behavior for the duration of the test set. This step allows users to make informed decisons about model performance and saves numerous costs associated with productionalizing the model and monitoring its performance in real life. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scripts.helper_scripts import _draw_one_plot\n",
```

Comment:
Nit: Let us rename from _draw_one_plot  to draw_one_plot as underscore means that it is a private function.

Reply:
done.